src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/LocalSCMListener.java

```diff
@@ -49,11 +47,8 @@ public void onCheckout(Run<?, ?> build, SCM scm, FilePath workspace, TaskListene
 
         //case 1 - bb_checkout step in the script (pipeline or groovy)
         if (scm instanceof BitbucketSCM) {
-            handleBitbucketSCMCheckout(build, (BitbucketSCM) scm, listener);
-            return;
-        }
-
-        if (isWorkflowRun(build)) {
+            handleBitbucketSCMCheckout(build, (BitbucketSCM) scm);
+        } else if (isWorkflowRun(build)) {
```

Comment:
Not changed in in this PR, but perhaps this could use [SCMSourceOwner.getSCMSources()](https://github.com/jenkinsci/scm-api-plugin/blob/0cbb94ba007882d4e60728d8282beed481bd671f/src/main/java/jenkins/scm/api/SCMSourceOwner.java#L39-L45) instead of MultiBranchProject\<?, ?>.getSources() and BranchSource.getSource, for greater flexibility. I don't know whether there are project types that could benefit from such a change, given how `isWorkflowRun` works.

[SCMSource.SourceByItem.findSource(Item)](https://github.com/jenkinsci/scm-api-plugin/blob/0cbb94ba007882d4e60728d8282beed481bd671f/src/main/java/jenkins/scm/api/SCMSource.java#L1241-L1256) may also be useful.